kirby/api/__init__.py

```diff
@@ -54,7 +62,19 @@ def _register(self, source_id=None, destination_id=None):
                 raise ClientError("There is an error with the id(s) given.")
 
     def add_source(self, source):
-        self._register(source_id=self.get_topic_id(source.name))
+        if not self.testing:
+            self._register(source_id=self.get_topic_id(source.name))
+        else:
+            logger.info(
+                f"add_source has been skipped, since app"
```

Comment:
if you're not using format, please don't prefix your strings with `f`

Reply:
Removed